common4j/src/main/com/microsoft/identity/common/java/controllers/results/SignUpCommandResult.kt

```diff
@@ -0,0 +1,109 @@
+ * @see com.microsoft.identity.common.java.providers.nativeauth.responses.signup.SignUpChallengeApiResult
+ * @see com.microsoft.identity.common.java.providers.nativeauth.responses.signup.SignUpContinueApiResult
+ */
+interface SignUpCommandResult {
+    data class UsernameAlreadyExists(
+        val error: String,
+        val errorDescription: String,
+        val correlationId: String = DiagnosticContext.INSTANCE.threadCorrelationId
+    ) : SignUpStartCommandResult, SignUpSubmitCodeCommandResult,
+        SignUpSubmitUserAttributesCommandResult, SignUpSubmitPasswordCommandResult
+
+    data class Complete (
+        val signInSLT: String?,
+        val expiresIn: Int?
+    ) : SignUpStartCommandResult,
+        SignUpSubmitCodeCommandResult, SignUpSubmitPasswordCommandResult,
+        SignUpSubmitUserAttributesCommandResult
+
+    data class CodeRequired(
+        val signupToken: String,
+        val challengeTargetLabel: String,
+        val challengeChannel: String,
+        val codeLength: Int
+    ) : SignUpStartCommandResult,
+        SignUpResendCodeCommandResult
+
+    data class PasswordRequired(
+        val signupToken: String
+    ) : SignUpStartCommandResult,
+        SignUpSubmitCodeCommandResult
+
+    data class AttributesRequired(
+        val signupToken: String,
+        val error: String,
+        val errorDescription: String,
+        val requiredAttributes: List<UserAttributeApiResult>,
+        val correlationId: String = DiagnosticContext.INSTANCE.threadCorrelationId
+    ) : SignUpStartCommandResult, SignUpSubmitPasswordCommandResult,
+        SignUpSubmitUserAttributesCommandResult,
+        SignUpSubmitCodeCommandResult
+
+    data class InvalidEmail(
+        val error: String,
+        val errorDescription: String,
+        val correlationId: String = DiagnosticContext.INSTANCE.threadCorrelationId
+    ) : SignUpStartCommandResult, SignUpSubmitPasswordCommandResult
+
+    data class InvalidPassword(
+        val error: String,
+        val errorDescription: String,
+        val correlationId: String = DiagnosticContext.INSTANCE.threadCorrelationId
+    ) : SignUpStartCommandResult, SignUpSubmitPasswordCommandResult
+
+    data class InvalidCode(
+        val error: String,
+        val errorDescription: String,
+        val correlationId: String = DiagnosticContext.INSTANCE.threadCorrelationId
+    ) : SignUpSubmitCodeCommandResult
+
+    data class InvalidAttributes(
+        val error: String,
+        val errorDescription: String,
+        val invalidAttributes: List<String>,
+        val correlationId: String = DiagnosticContext.INSTANCE.threadCorrelationId
+    ) : SignUpStartCommandResult, SignUpSubmitUserAttributesCommandResult
+
+    data class AuthNotSupported(
+        val error: String,
+        val errorDescription: String,
+        val correlationId: String = DiagnosticContext.INSTANCE.threadCorrelationId
```

Comment:
Just a comment to make note of correlationID

Reply:
I have added a comment for correlation id